### PR TITLE
Fix 500 responses from v0 metrics API

### PR DESCRIPTION
**Pull request by branden**
## High-level description

This PR bumps the Telegraf package to pull in a fix for an issue that causes some requests to the v0 metrics API to get 500 responses due to a JSON serialization error.

## Corresponding DC/OS tickets (obligatory)

These DC/OS JIRA ticket(s) must be updated (ideally closed) in the moment this PR lands:

  - [DCOS_OSS-4088](https://jira.mesosphere.com/browse/DCOS_OSS-4088) metrics v0 API 500 error: "Failed to encode value to JSON"

## Related tickets (optional)

Other tickets related to this change:

N/A

## Checklist for all PRs

  - [x] Added a comprehensible changelog entry to `CHANGES.md` or explain why this is not a user-facing change: This fixes a regression that hasn't yet been released.
  - [x] Included a test which will fail if code is reverted but test is not. If there is no test please explain here: 
  - [x] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [x] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [x] Change log from the last version integrated (this should be a link to commits for easy verification and review): https://github.com/dcos/telegraf/compare/ccb5eb8c53842b690ace1af2c40ff6101e0dc93e...071b143b1c93f03e5469378d082ec14dacf41674
  - [x] Test Results: [link to CI job test results for component](https://jenkins.mesosphere.com/service/jenkins/blue/organizations/jenkins/public-dcos-cluster-ops%2Ftelegraf%2Ftelegraf-dcos/detail/telegraf-dcos/40/pipeline)
  - [x] Code Coverage (if available): [link to code coverage report](https://jenkins.mesosphere.com/service/jenkins/blue/organizations/jenkins/public-dcos-cluster-ops%2Ftelegraf%2Ftelegraf-dcos/detail/telegraf-dcos/40/pipeline)

### CODE REVIEW

**Comment by branden** [comment]
@mesosphere-mergebot bump-ee

**Comment by branden** [comment]
@mesosphere-mergebot bump-ee

**Comment by branden** [comment]
@mesosphre-mergebot bump-ee

**Comment by branden** [comment]
@mesosphere-mergebot bump-ee

**Comment by branden** [comment]
@mesosphere-mergebot bump-ee

**Comment by branden** [comment]
Test failures look unrelated.

* `teamcity/dcos/test/azure/arm`
  * https://jira.mesosphere.com/browse/DCOS-41677
* `teamcity/dcos/test/dcos-docker/static`
  * https://jira.mesosphere.com/browse/DCOS-20756
* `teamcity/dcos/test/aws/cloudformation/simple`
  * This error is new to me. All tests passed, but log collection failed: https://teamcity.mesosphere.io/viewLog.html?tab=buildLog&logTab=tree&filter=debug&expand=all&buildId=1209147&_focus=6692
* `teamcity/dcos/test/aws/onprem/static`
  * https://jira.mesosphere.com/browse/DCOS_OSS-2115

**Comment by branden** [comment]
@mesosphere-mergebot label ready for review

File: packages/dcos-integration-test/extra/test_metrics.py [hunk 1]
```diff
@@ -248,3 +221,103 @@ def wait_for_container_response():
...
+    task_name = 'test-metrics-containers-nan'
+    metric_name = 'test_metrics_containers_nan'
+    marathon_app = {
+        'id': '/' + task_name,
+        'instances': 1,
+        'cpus': 0.1,
+        'mem': 128,
+        'cmd': '\n'.join([
+            'echo "Sending gauge with NaN value to $STATSD_UDP_HOST:$STATSD_UDP_PORT"',
+            'echo "{}:NaN|g" | nc -w 1 -u $STATSD_UDP_HOST $STATSD_UDP_PORT'.format(metric_name),
```
Comment (klueska) [inline packages/dcos-integration-test/extra/test_metrics.py]: You shouldn't need the `-w 1` if you are only sending this once.

File: packages/dcos-integration-test/extra/test_metrics.py [hunk 2]
```diff
@@ -248,3 +221,103 @@ def wait_for_container_response():
...
+        assert len(dps) == 1, 'expected 1 datapoint for metric {}, got {}'.format(dp_name, len(dps))
+        return dps[0]['value']
+
+    with dcos_api_session.marathon.deploy_and_cleanup(marathon_app, check_health=False):
+        endpoints = dcos_api_session.marathon.get_app_service_endpoints(marathon_app['id'])
+        assert len(endpoints) == 1, 'The marathon app should have been deployed exactly once.'
+        node = endpoints[0].host
+
+        # NaN should be converted to empty string.
+        metric_value = _get_app_metric_datapoint_value_for_task(dcos_api_session, node, task_name, metric_name)
```
Comment (klueska) [inline packages/dcos-integration-test/extra/test_metrics.py]: Did we finally decide that empty string was the right value for NaN?

File: packages/dcos-integration-test/extra/test_metrics.py (hunk 1 above)
Comment (branden) [reply packages/dcos-integration-test/extra/test_metrics.py]: `-w` sets the connect/read timeout. Is that relevant to the number of times the message is sent?

File: packages/dcos-integration-test/extra/test_metrics.py (hunk 2 above)
Comment (branden) [reply packages/dcos-integration-test/extra/test_metrics.py]: Yeah

File: packages/dcos-integration-test/extra/test_metrics.py (hunk 1 above)
Comment (klueska) [reply packages/dcos-integration-test/extra/test_metrics.py]: I know I only added it in the script you pulled this from to support calling `nc` multiple times. That said, it doesn't hurt to have it there as a timeout.

**Comment by klueska** [comment]
There still seems to be alot of failing tests.

**Comment by branden** [comment]
@klueska See https://github.com/dcos/dcos/pull/3436#issuecomment-421528056

**Comment by orsenthil** [comment]
@mesosphere-mergebot override-status teamcity/dcos/test/aws/onprem/static https://jira.mesosphere.com/browse/DCOS_OSS-2115

**Comment by cprovencher** [comment]
@mesosphere-mergebot override-status teamcity/dcos/test/aws/cloudformation/simple

**Comment by cprovencher** [comment]
@mesosphere-mergebot override-status teamcity/dcos/test/aws/cloudformation/simple https://jira.mesosphere.com/browse/DCOS-41751

**Comment by orsenthil** [comment]
@mesosphere-mergebot override-status teamcity/dcos/test/azure/arm https://jira.mesosphere.com/browse/DCOS-41677

**Comment by orsenthil** [comment]
@mesosphere-mergebot override-status teamcity/dcos/test/dcos-docker/static DCOS-20756

**Comment by orsenthil** [comment]
@mesosphere-mergebot override-status teamcity/dcos/build/dcos/windows DCOS-41386

**Comment by jgehrcke** [comment]
@mesosphere-mergebot override-status teamcity/dcos/build/dcos/windows https://jira.mesosphere.com/browse/DCOS-42046

**Comment by jgehrcke** [comment]
@mesosphere-mergebot pr-status-check

**Pull request merged**